File: IBCMAndroid/app/src/main/java/com/example/ibcmserver_init/data/model/Report.kt
data class Report(
    val id: String,
    val type: String, // "EVENT", "USER", "CHAT", "MESSAGE"
    val targetId: String,
    val reason: String,
    val reporterId: String,
    val timestamp: Long,
    val status: String, // "PENDING", "REVIEWED", "RESOLVED", "REJECTED"
    val reviewerNotes: String? = null,
    val reviewedAt: Long? = null,
    val reviewerId: String? = null
) 

File: IBCM-ios/IBCM/IBCM/Models/Report.swift
import Foundation

struct Report: Codable {
    let id: String
    let targetId: String
    let targetType: String // "user", "event", "comment", etc.
    let reason: String
    let description: String
    let reporterId: String
    let status: String // "pending", "resolved", "rejected"
    let createdAt: Date
    let updatedAt: Date
    let evidence: [String]? // URLs to evidence (screenshots, etc.)
}

extension Report {
    static let validTargetTypes = ["user", "event", "comment", "message", "product", "review"]
    static let validReasons = ["spam", "harassment", "inappropriate", "hate_speech", "violence", "fraud", "impersonation", "other"]
    static let validStatuses = ["pending", "resolved", "rejected"]
} 